src/NEPTypes.jl

```diff
@@ -570,11 +568,50 @@ julia> compute_Mder(pep,3)-(A0+A1*3+A2*9)
         for i=1:size(nep.A,1)
             a[i]=dot(y,nep.A[i]*x);
         end
-        p=Poly(a);
-        rr=real(poly_roots(p));  # Only works if polynomial roots are real
-        return rr
+        m=size(nep.A,1);
+        # Build a bigfloat companion matrix
+        A=zeros(T,m-1,m-1);
+        for k=1:m-1
+            if (k<m-1)
+                A[k,k+1]=1;
+            end
+            A[end,k]=-a[k]/a[end]
+        end
+        pp=eigvals(A);
+        if (T <: Real) # If we specify real arithmetic, return real eigvals
+            pp=real(pp);
+        end
+
+        II=sortperm(abs.(pp .- target))
+        return pp[II];
     end
 
+    # Overload a version of eigvals(::Matrix{BigFloat}) for compute_rf
+    import LinearAlgebra.eigvals
+    function eigvals(A::Union{Matrix{BigFloat},Matrix{Complex{BigFloat}}})
+        T=eltype(A);
+        Tfloat= (T <: Complex) ? (ComplexF64) : (Float64)
+        Afloat=Matrix{Tfloat}(A);
+        λv,X=eigen(Afloat);
+        local λv_bigfloat=Vector{Complex{BigFloat}}(λv);
+        # Some Rayleigh quotient iteration steps in bigfloat to improve accuracy:
+        for j=1:size(λv,1)
+            local s=λv_bigfloat[j];
+            z=Vector{Complex{BigFloat}}(X[:,j])
+            for f=1:3
```

Comment:
Could the number of iterations be figured out dynamically, by residual norm or something? (Or perhaps there's no need for that?)

Reply:
In principle yes, but they are always heuristic. The decision to take three is however already from theory. The Rayleigh quotient iteration has cubic convergence. If we initiate with an approximation with error 10^-16:

Error after one step (10^-16)^3=10^-49 
Error after two steps (10^-49)^3=10^-145 which is smaller than eps(BigFloat)=10^-77. 

I take one extra step to be "sure".